lib/ansible/modules/commands/psexec.py

```diff
@@ -0,0 +1,527 @@
+  type: str
+  sample: Error [10] running process
+pid:
+  description: The process ID of the asynchronous process that was created
+  returned: success and asynchronous is 'yes'
+  type: int
+  sample: 719
+rc:
+  description: The return code of the remote process
+  returned: success and asynchronous is 'no'
+  type: int
+  sample: 0
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_bytes, to_native, to_text
+
+try:
+    from pypsexec import client
+    from pypsexec.exceptions import PypsexecException, PAExecException, \
+        PDUException, SCMRException
+    from pypsexec.paexec import ProcessPriority
+    from smbprotocol.exceptions import SMBException, SMBAuthenticationError, \
+        SMBResponseException
+    HAS_PYPSEXEC = True
+except ImportError:
+    HAS_PYPSEXEC = False
+
+try:
+    import gssapi
+    # GSSAPI extension required for Kerberos Auth in SMB
+    from gssapi.raw import inquire_sec_context_by_oid
+    HAS_KERBEROS = True
+except ImportError:
+    HAS_KERBEROS = False
+
+
+def remove_artifacts(module, client):
+    warning_msg = "Failed to cleanup PAExec service and executable"
+    try:
+        client.remove_service()
+    except PDUException as exc:
+        module.warn("%s with RPC PDU fault: %s" % (warning_msg, to_text(exc)))
+    except SCMRException as exc:
+        module.warn("%s with an SCMR fault: %s" % (warning_msg, to_text(exc)))
+    except PypsexecException as exc:
+        module.warn("%s with generic fault: %s" % (warning_msg, to_text(exc)))
+    except SMBException as exc:
+        module.warn("Failed to remove PAExec executable with SMB fault: %s"
+                    % to_text(exc))
+
+
+def main():
+    module_args = dict(
+        hostname=dict(type='str', required=True),
+        connection_username=dict(type='str'),
+        connection_password=dict(type='str', no_log=True),
+        port=dict(type='int', required=False, default=445),
+        encrypt=dict(type='bool', default=True),
+        connection_timeout=dict(type='int', default=60),
+        executable=dict(type='str', required=True),
+        arguments=dict(type='str'),
+        working_directory=dict(type='str', default='C:\\Windows\\System32'),
+        asynchronous=dict(type='bool', default=False),
+        load_profile=dict(type='bool', default=True),
+        process_username=dict(type='str'),
+        process_password=dict(type='str', no_log=True),
+        elevated=dict(type='bool', default=False),
+        limited=dict(type='bool', default=False),
+        use_system=dict(type='bool', default=False),
+        interactive=dict(type='bool', default=False),
+        interactive_session=dict(type='int', default=0),
+        priority=dict(type='str', default='normal',
+                      choices=['above_normal', 'below_normal', 'high',
+                               'idle', 'normal', 'realtime']),
+        show_ui_on_win_logon=dict(type='bool', default=False),
```

Comment:
Feels like this could use a better name, but I'm not sure what... `show_ui_on_secure_desktop`? `show_ui_on_logon_screen`?

Reply:
I think I like `show_ui_on_logon_screen`, people would understand it better.